File: samples/MinimalZeroInstall.cs
using NanoByte.Common.Tasks;
using ZeroInstall.Model;
using ZeroInstall.Services;

var requirements = new Requirements(new FeedUri(args[0]));
var services = new ServiceProvider(new CliTaskHandler());
var selections = services.Solver.Solve(requirements);
foreach (var implementation in services.SelectionsManager.GetUncached(selections.Implementations))
    services.Fetcher.Fetch(implementation);
services.Executor.Start(selections);

File: samples/MinimalZeroInstall.vb
Imports NanoByte.Common.Tasks
Imports ZeroInstall.Model
Imports ZeroInstall.Services

Module MinimalZeroInstall
    Sub Main(ByVal args As String())
        Dim requirements = New Requirements(New FeedUri(args(0)))
        Dim services = New ServiceProvider(New CliTaskHandler())
        With services
            Dim selections = .Solver.Solve(requirements)
            For Each implementation In .SelectionsManager.GetUncachedImplementations(selections)
                .Fetcher.Fetch(implementation)
            Next
            .Executor.Start(selections)
        End With
    End Sub
End Module
